src/ir/possible-contents.cpp

```diff
@@ -1016,6 +1016,23 @@ struct InfoCollector
     addChildParentLink(curr->ref, curr);
     addChildParentLink(curr->value, curr);
   }
+  void visitStructRMW(StructRMW* curr) {
+    if (curr->ref->type == Type::unreachable) {
+      return;
+    }
+    // Similar to visitStructSet.
+    addChildParentLink(curr->ref, curr);
+    addChildParentLink(curr->value, curr);
+  }
+  void visitStructCmpxchg(StructCmpxchg* curr) {
+    if (curr->ref->type == Type::unreachable) {
+      return;
+    }
+    // Similar to visitStructSet.
+    addChildParentLink(curr->ref, curr);
+    addChildParentLink(curr->expected, curr);
+    addChildParentLink(curr->replacement, curr);
+  }
   // Array operations access the array's location, parallel to how structs work.
```

Comment:
I suspect these need to be added on line 2369, like StructGet is, for packed field reads.

Reply:
`filterPackedDataReads` only does anything for signed packed reads, but atomic accesses always do unsigned reads, so I don't think any changes are necessary here. I'll add a comment, though.